### ISSUE TITLE
py3.10 issue

### ISSUE THREAD

**Issue by dragonyanglong** [diffpy/diffpy.pdffit2#14]
when run pdfgui in py3.10, the error shows:

```
SystemError: PY_SSIZE_T_CLEAN macro must be defined for '#' formats

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\simon\miniconda3\envs\pdfgui310-new\lib\site-packages\diffpy\pdfgui\gui\errorwrapper.py", line 62, in _f
    return func(*args, **kwargs)
  File "C:\Users\simon\miniconda3\envs\pdfgui310-new\lib\site-packages\diffpy\pdfgui\gui\mainframe.py", line 1884, in onRun
    self.control.start(IDlist)
  File "C:\Users\simon\miniconda3\envs\pdfgui310-new\lib\site-packages\diffpy\pdfgui\control\pdfguicontrol.py", line 498, in start
    calc.start()
  File "C:\Users\simon\miniconda3\envs\pdfgui310-new\lib\site-packages\diffpy\pdfgui\control\calculation.py", line 125, in start
    self.calculate()
  File "C:\Users\simon\miniconda3\envs\pdfgui310-new\lib\site-packages\diffpy\pdfgui\control\calculation.py", line 183, in calculate
    server.calc()
  File "C:\Users\simon\miniconda3\envs\pdfgui310-new\lib\site-packages\diffpy\pdffit2\pdffit.py", line 315, in calc
    pdffit2.calc(self._handle)
SystemError: <built-in function calc> returned a result with an exception set
```

We have tried to solve it by add `#define PY_SSIZE_T_CLEAN`, see c62e0e3 and 72481c0 commits. But seems like it still fails?

**Comment by dragonyanglong** [comment]
@sbillinge this is the issue

**Comment by sbillinge** [comment]
There is some discussion of this here:
https://stackoverflow.com/questions/70705404/systemerror-py-ssize-t-clean-macro-must-be-defined-for-formats

**Comment by 8bitsam** [comment]
I can't seem to replicate this issue. It might be related to the packages not being updated on the user side. I would try running `conda update -c conda-forge diffpy.pdffit2` and `conda update -c conda-forge diffpy.pdfgui` and see if it still persists, then try again in a new conda environment with a fresh install of both pdffit2 and pdfgui.

**Comment by sbillinge** [comment]
closing as seemingly fixed by #11
